### PR TITLE
generators: add --no-branch option to prevent creating new local branch (#248)

### PR DESCRIPTION
**Pull request by shr-project**
* useful when regenerating the files locally by hand without planing to submit
  PR with the changes or when some other tool will do additional
  integration before doing the PR (like we plan to do with meta-ros)

Signed-off-by: Martin Jansa <martin.jansa@lge.com>

### CODE REVIEW

File: superflore/generators/bitbake/run.py [hunk 1]
```diff
@@ -93,7 +93,8 @@ def main():
         overlay = RosMeta(
             _repo,
             not args.output_repository_path,
-            branch='superflore/{}'.format(now),
+            branch=(('superflore/{}'.format(now)) if not args.no_branch
+                                                  else None),
```
Comment (allenh1) [inline superflore/generators/bitbake/run.py]: @shr-project the linter is unhappy with this line

Comment (shr-project) [reply superflore/generators/bitbake/run.py]: Looking into it now (before the line was too long now this.. :))

**Comment by shr-project** [comment]
Fixes #248

File: superflore/generators/bitbake/run.py (hunk 1 above)
Comment (allenh1) [reply superflore/generators/bitbake/run.py]: > before the line was too long now this.. :)

Haha, yeah. Welcome to superflore changes ;). Linter is pretty picky.

**Comment by shr-project** [comment]
Hi @allenh1 is there anything blocking this PR? We're almost ready to publish updated meta-ros with recipes generated with superflore modified with all these open PRs and it would be nice to have it in superflore/master before we do.

File: superflore/parser.py [hunk 2]
```diff
@@ -40,6 +40,11 @@ def get_parser(tool_tip, is_generator=True, exclude_all=False):
             help='ONLY file a PR to remote',
             action='store_true'
         )
+        parser.add_argument(
+            '--no-branch',
+            help='Do not create new branch automatically',
```
Comment (allenh1) [inline superflore/parser.py]:
```suggestion
            help='Do not create a new branch automatically',
```

File: README.md [hunk 3]
```diff
@@ -155,6 +156,7 @@ optional arguments:
                         regenerate packages for the specified distro
   --dry-run             run without filing a PR to remote
   --pr-only             ONLY file a PR to remote
+  --no-branch           don't create new local branch
```
Comment (allenh1) [inline README.md]:
```suggestion
  --no-branch           Do not create a new branch automatically
```

File: README.md [hunk 4]
```diff
@@ -48,6 +48,7 @@ optional arguments:
   --all                 regenerate all packages in all distros
   --dry-run             run without filing a PR to remote
   --pr-only             ONLY file a PR to remote
+  --no-branch           don't create new local branch
```
Comment (allenh1) [inline README.md]:
```suggestion
  --no-branch           Do not create a new branch automatically
```

File: superflore/generators/ebuild/run.py [hunk 5]
```diff
@@ -85,6 +85,7 @@ def main():
             org=repo_org,
             repo=repo_name,
             from_branch=args.upstream_branch,
+            new_branch=not args.no_branch,
```
Comment (allenh1) [inline superflore/generators/ebuild/run.py]:
```suggestion
            new_branch=(not args.no_branch),
```

**Pull request merged**